### PR TITLE
[Kotlin Migration] AsyncDialogFragment

### PR DESCRIPTION
**Pull request by Akshay0701**

### CODE REVIEW

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
```

e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.kt: (36, 1): Class 'DatabaseErrorDialog' is not abstract and does not implement abstract base class member public abstract val notificationMessage: String? defined in com.ichi2.anki.dialogs.AsyncDialogFragment
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.kt: (386, 5): 'getNotificationMessage' overrides nothing
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.kt: (390, 5): 'getNotificationTitle' overrides nothing
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.kt: (394, 5): 'getDialogHandlerMessage' overrides nothing
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/ExportCompleteDialog.kt: (25, 1): Class 'ExportCompleteDialog' is not abstract and does not implement abstract base class member public abstract val notificationMessage: String? defined in com.ichi2.anki.dialogs.AsyncDialogFragment
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/ExportCompleteDialog.kt: (46, 20): Type mismatch: inferred type is String? but CharSequence was expected
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/ExportCompleteDialog.kt: (47, 22): Type mismatch: inferred type is String? but CharSequence was expected
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/ExportCompleteDialog.kt: (64, 5): 'getNotificationTitle' overrides nothing
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/ExportCompleteDialog.kt: (68, 5): 'getNotificationMessage' overrides nothing
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/ImportDialog.kt: (26, 1): Class 'ImportDialog' is not abstract and does not implement abstract base class member public abstract val notificationMessage: String? defined in com.ichi2.anki.dialogs.AsyncDialogFragment
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/ImportDialog.kt: (80, 5): 'getNotificationMessage' overrides nothing
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/ImportDialog.kt: (84, 5): 'getNotificationTitle' overrides nothing
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/MediaCheckDialog.kt: (18, 1): Class 'MediaCheckDialog' is not abstract and does not implement abstract base class member public abstract val notificationMessage: String? defined in com.ichi2.anki.dialogs.AsyncDialogFragment
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/MediaCheckDialog.kt: (30, 23): Type mismatch: inferred type is String? but CharSequence was expected
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/MediaCheckDialog.kt: (33, 33): Type mismatch: inferred type is String? but CharSequence was expected
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/MediaCheckDialog.kt: (117, 5): 'getNotificationMessage' overrides nothing
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/MediaCheckDialog.kt: (125, 5): 'getNotificationTitle' overrides nothing
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/MediaCheckDialog.kt: (131, 5): 'getDialogHandlerMessage' overrides nothing
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SimpleMessageDialog.kt: (27, 1): Class 'SimpleMessageDialog' is not abstract and does not implement abstract base class member public abstract val notificationMessage: String? defined in com.ichi2.anki.dialogs.AsyncDialogFragment
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SimpleMessageDialog.kt: (50, 5): 'getNotificationTitle' overrides nothing
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SimpleMessageDialog.kt: (59, 5): 'getNotificationMessage' overrides nothing
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SyncErrorDialog.kt: (31, 1): Class 'SyncErrorDialog' is not abstract and does not implement abstract base class member public abstract val notificationMessage: String? defined in com.ichi2.anki.dialogs.AsyncDialogFragment
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SyncErrorDialog.kt: (200, 5): 'getNotificationTitle' overrides nothing
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SyncErrorDialog.kt: (227, 5): 'getNotificationMessage' overrides nothing
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/SyncErrorDialog.kt: (233, 5): 'getDialogHandlerMessage' overrides nothing
```

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
Small nitpick, sorry

**Comment by Akshay0701** [comment]
yeah sure, tell me what is the nitpick?

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/AsyncDialogFragment.kt [hunk 1]
```diff
@@ -13,35 +13,31 @@
...
        This can happen when the DialogFragment is shown from 
        the onPostExecute() method of an AsyncTask */
-    
-    public abstract String getNotificationMessage();
-    public abstract String getNotificationTitle();
-
-    public Message getDialogHandlerMessage() {
-        return null;
+    abstract fun getNotificationMessage(): String?
+    abstract fun getNotificationTitle(): String?
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/AsyncDialogFragment.kt:30]: Really sorry, comment didn't go through.

Could you add a `@KotlinCleanup` to convert these back to properties

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
Comment added

**Review by david-allison-1 — APPROVED** [review]
LGTM

**Review by mikehardy — APPROVED** [review]
Absolutely massive batch @Akshay0701 - thank you!

**Pull request merged**